AutoCollection/Statsbeat.ts

```diff
@@ -54,7 +54,7 @@ class Statsbeat {
         let statsbeatConnectionString = this._getConnectionString(config);
         this._statsbeatConfig = new Config(statsbeatConnectionString);
         this._statsbeatConfig.samplingPercentage = 100; // Do not sample
-        this._sender = new Sender(this._statsbeatConfig, null, null, this._handleNetworkError);
+        this._sender = new Sender(this._statsbeatConfig, null, null, null, null, true, this._shutdownStatsbeat);
```

Comment:
parameters are getting jammed. how do you handle that in node? what is the limit or keep growing?

Reply:
When the object starts doing stuff the moment is instantiated, we need to provide the settings in the constructor, it could receive a single "settings" object with everything instead of multiple required and optional parameters, that is what I prefer to do but that will be a breaking change so we cannot do it right now, customer could be instantiating this class directly so we cannot change the signature now.